internal/util/telemetry/telemetry.go

```diff
@@ -60,15 +60,18 @@ func (s *Flag) UnmarshalText(text []byte) error {
 //   - common DO_NOT_TRACK environment variable;
 //   - executable name;
 //   - and the previously saved state.
-func initialState(f *Flag, dnt string, execName string, prev *bool, l *zap.Logger) (*bool, error) {
+//
+// The second returned value is true if the telemetry state should be locked, because of
+// setting telemetry via a command-line flag, an environment variable, or a filename.
+func initialState(f *Flag, dnt string, execName string, prev *bool, l *zap.Logger) (*bool, bool, error) {
 	var disable bool
 
 	// https://consoledonottrack.com is not entirely clear about accepted values.
 	// Assume that "1", "t", "true", etc. mean that telemetry should be disabled,
 	// and other valid values, including "0" and empty string, mean undecided.
 	v, err := parseValue(dnt)
 	if err != nil {
-		return nil, err
+		return nil, false, err
```

Comment:
missing test?

Reply:
It was already there so I missed that one, but sure this could be covered with a test, nice catch!